arrow-core-data/src/main/kotlin/arrow/typeclasses/Foldable.kt

```diff
@@ -205,22 +205,23 @@ interface Foldable<F> {
   /**
    * Get the first element of the foldable or none
    */
-  fun <A> Kind<F, A>.firstOption(): Option<A> = get(0)
+  fun <A> Kind<F, A>.firstOption(): Option<A> =
+    find { true }
 
   /**
-   * Get the first element of the foldable or none if empty or the predicate does not match
+   * Get the first element matching the predicate or none
    */
   fun <A> Kind<F, A>.firstOption(predicate: (A) -> Boolean): Option<A> =
```

Comment:
btw, don't we have a test to verify this? IMO either the tests should either have failed or we should change them to verify this

Reply:
`Foldable` does have two tests but they only concern the two cases `F` is emtpy and the predicate always returns false. This does not cover the multiple elements match the predicate case.

Reply:
Should we merge this change then and wait for the new split of `arrow-test` to extend the laws?
I've created a [new issue for it](https://github.com/arrow-kt/arrow-core/issues/34)

Reply:
I'm fine with merging this as is. Is `find` covered by any tests? If so, the missing test case won't matter until we think about covering derived functionality with tests anyway.

Reply:
Sort of.. there is one law to check wether `exists` is consistent with `find`.. so definitely `FoldableLaws` test suite needs to be extended

Reply:
Hmm ok... I'm fine with merging anyway, as this is quite a bit more useful than before^^

`FoldableLaws` definitely needs work, quite a bit actually because with `Foldable` local overrides in specific instances (think specific search implementations for trees etc) make a ton of sense.
I am starting to dislike the term `Foldable*Laws*` though, because these are more tests than laws, but thats nitpicking and doesn't matter really :sweat_smile: